### DISCUSSION TITLE
declare features in machine learning trainig

### DISCUSSION THREAD

**Discussion by mahdishemshadi** [kernc/backtesting.py#261] [Q&A]
Hi,
How can I declare features in machine learning training code?
I mean in this link
https://kernc.github.io/backtesting.py/doc/examples/Trading%20with%20Machine%20Learning.html

**Comment by jjphung** [comment]
I assume your features are indicators or signals. There are tutorials covering this already. [Indicators](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html) and [Signals](https://kernc.github.io/backtesting.py/doc/examples/Strategies%20Library.html)
